controllers/subscription.ts

```diff
@@ -0,0 +1,81 @@
+import { CustomRequest, CustomResponse } from "../types/global";
+const { addOrUpdate } = require("../models/users");
+const { INTERNAL_SERVER_ERROR } = require("../constants/errorMessages");
+const nodemailer = require("nodemailer");
+const config = require("config");
+const emailSubscriptionCredentials = config.get("emailSubscriptionCredentials");
+
+export const subscribe = async (req: CustomRequest, res: CustomResponse) => {
+  try {
+    const { email } = req.body;
+    const phoneNumber = req.body.phoneNumber || null;
+    const userId = req.userData.id;
+    const dev = req.query.dev === "true";
+    if (!dev) {
+      return res.boom.notFound("Route not found");
+    }
+
+    const data = { email, isSubscribed: true, phoneNumber };
+    await addOrUpdate(data, userId);
+    return res.status(201).json({
+      message: "user subscribed successfully",
+    });
+  } catch (error) {
+    res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+export const unsubscribe = async (req: CustomRequest, res: CustomResponse) => {
+  try {
+    const userId = req.userData.id;
+    const dev = req.query.dev === "true";
+    if (!dev) {
+      return res.boom.notFound("Route not found");
+    }
+    await addOrUpdate(
+      {
+        isSubscribed: false,
+      },
+      userId
+    );
+    return res.status(200).json({
+      message: "user unsubscribed successfully",
+    });
+  } catch (error) {
+    res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+export const sendEmail = async (req: CustomRequest, res: CustomResponse) => {
```

Comment:
Should this be in a separate file?

Reply:
Yes this will be separate file, but for the time being we are only sending email to one person that's why I am using this here. Tough I'll add a TODO here.